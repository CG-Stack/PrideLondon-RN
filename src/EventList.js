// @flow
import React from "react";
import { View, StyleSheet, FlatList } from "react-native";
import EventCard from "./EventCard";
import type { Event } from "./integrations/cms";

type Props = {
  locale: string,
  events: Event[],
  refreshing: boolean,
  onRefresh: () => void
};

const EventList = ({ events, locale, refreshing, onRefresh }: Props) => (
  <FlatList
    data={events}
    keyExtractor={event => event.sys.id}
    renderItem={({ item: event }) => (
      <View style={styles.eventListItem}>
        <EventCard name={event.fields.name[locale]} />
      </View>
    )}
    refreshing={refreshing}
    onRefresh={onRefresh}
  />
);

const styles = StyleSheet.create({
  eventListItem: {
    paddingLeft: 15,
    paddingRight: 15,
    paddingBottom: 10
  }
});

export default EventList;
