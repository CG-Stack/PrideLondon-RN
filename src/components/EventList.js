// @flow
import React from "react";
import { StyleSheet, SectionList, TouchableOpacity, View } from "react-native";
import type { SectionBase } from "react-native/Libraries/Lists/SectionList";
import formatDate from "date-fns/format";

import EventCard from "./EventCard";
import Text from "./Text";
import type { Event, EventDays, Asset } from "../data/event";
import {
  bgColor,
  eventCardTextColor,
  sectionHeaderShadow,
  sectionHeaderBgColor,
  eventCardShadow
} from "../constants/colors";

type Props = {
  locale: string,
  events: EventDays,
  refreshing?: boolean,
  onRefresh?: () => void,
  onPress: (eventName: string) => void,
  getAssetById: string => Asset
};

const separator = style => () => <View style={style} />;

type ItemProps = { item: Event };
const renderItem = (styles, locale, onPress, getAssetById) => ({
  item: event
}: ItemProps) => (
  <TouchableOpacity
    style={styles.eventListItem}
    accessibilityTraits={["button"]}
    accessibilityComponentType="button"
    delayPressIn={50}
    onPress={() => onPress(event.sys.id)}
  >
    <EventCard
      name={event.fields.name[locale]}
      locationName={event.fields.locationName[locale]}
      eventPriceLow={event.fields.eventPriceLow[locale]}
      eventPriceHigh={event.fields.eventPriceHigh[locale]}
      startTime={event.fields.startTime[locale]}
      endTime={event.fields.endTime[locale]}
      imageUrl={`https:${
        getAssetById(event.fields.eventsListPicture[locale].sys.id).fields.file[
          locale
        ].url
      }`}
      isFree={event.fields.isFree[locale]}
    />
  </TouchableOpacity>
);

type Section = SectionBase<Event> & { title: string };

type SectionProps = { section: Section };
const renderSectionHeader = styles => ({ section }: SectionProps) => (
  <Text type="h2" style={styles.sectionHeader}>
    {section.title}
  </Text>
);

const eventSections = (events: EventDays, locale: string): Section[] =>
  events.map(it => ({
    data: it,
    title: formatDate(it[0].fields.startTime[locale], "dddd D MMMM")
  }));

const EventList = ({
  events,
  locale,
  refreshing,
  onRefresh,
  onPress,
  getAssetById
}: Props) => (
  <SectionList
    stickySectionHeadersEnabled
    sections={eventSections(events, locale)}
    renderSectionHeader={renderSectionHeader(styles)}
    renderSectionFooter={separator(styles.sectionFooter)}
    renderItem={renderItem(styles, locale, onPress, getAssetById)}
    keyExtractor={event => event.sys.id}
    contentContainerStyle={styles.container}
    ItemSeparatorComponent={separator(styles.itemSeparator)}
    SectionSeparatorComponent={separator(styles.sectionSeparator)}
    refreshing={refreshing}
    onRefresh={onRefresh}
  />
);

EventList.defaultProps = {
  refreshing: false,
  onRefresh: undefined
};

const styles = StyleSheet.create({
  itemSeparator: {
    height: 12
  },
  sectionSeparator: {
    height: 6
  },
  container: {
    paddingTop: 0,
    backgroundColor: bgColor
  },
  eventListItem: {
    marginHorizontal: 16,
    borderRadius: 5,
    // The below properties are required for ioS shadow
    shadowColor: eventCardShadow,
    shadowOffset: { width: 0, height: 1 },
    shadowOpacity: 1,
    shadowRadius: 3,
    // The below properties are required for android shadow
    borderWidth: 0,
    elevation: 3,
    backgroundColor: bgColor
  },
  sectionHeader: {
    height: 40,
    color: eventCardTextColor,
    paddingVertical: 8,
    paddingHorizontal: 15,
    backgroundColor: sectionHeaderBgColor,

    // The below properties are required for ioS shadow
    shadowColor: sectionHeaderShadow,
    shadowOffset: { width: 0, height: 1 },
    shadowOpacity: 1,
    shadowRadius: 3,
    // The below properties are required for android shadow
    borderWidth: 0,
    elevation: 3,
    marginBottom: 6
  },
  sectionFooter: {
    height: 6
  }
});

export default EventList;
