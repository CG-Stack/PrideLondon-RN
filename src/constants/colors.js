export const bgColor = "#808080";
export const cardBgColor = "#FFFFFF";
export const imageBgColor = "#C1C1C1";
export const headerBgColor = "#A5A5A5";
export const priceTagBgColor = "#989898";

export const textColor = "#272727";

export const dialogBackdropColor = "rgba(0, 0, 0, 0.7)";
export const dialogBgColor = "#FFFFFF";
export const dialogApplyButtonBgColor = "rgb(51, 51, 51)";
export const dialogApplyButtonTextColor = "#FFFFFF";
export const dialogHeaderDividerColor = "#F2F2F2";

// FILTER HEADER

export const interestButtonBgColor = "rgb(32, 33, 90)";
export const interestButtonTextColor = "rgb(44, 218, 157)";
export const filterButtonBorderColor = "rgb(44, 218, 157)";
export const filterButtonTextColor = "rgb(44, 218, 157)";

// EVENT LIST

export const eventListBgColor = "rgb(226, 226, 226)";
export const eventListHeaderBgColor = "rgb(216, 216, 216)";
export const eventListHeaderColor = "rgb(51, 51, 51)";
export const lightNavyBlue = "rgb(45, 47, 127)";
export const lightGrey = "rgb(243, 243, 243)";
export const white = "rgb(255, 255, 255)";
export const eventCardShadow = "rgba(0, 0, 0, 0.2)";
export const sectionHeaderShadow = "rgba(0, 0, 0, 0.14)";
export const eucalyptusGreen = "rgb(44, 218, 157)";
export const darkBlueGreyTwo = "rgb(32, 33, 90)";

// EVENT DETAILS
export const eventDetailsBgColor = "#FFF";
export const eventDetailsHeaderBgColor = "#F2F2F2";
export const eventCategoryLabelBgColor = "#D8D8D8";
export const eventDetailsBuyButtonColor = "#D8D8D8";
export const eventDetailsBuyButtonTextColor = "#FFFFFF";
