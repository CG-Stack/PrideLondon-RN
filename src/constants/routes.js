export const EVENT_DETAILS = "EVENT_DETAILS";
export const EVENT_LIST = "EVENT_LIST";
export const FEATURED_EVENT_LIST = "FEATURED_EVENT_LIST";
export const HOME = "HOME";
export const EVENT_CATEGORY_FILTER = "EVENT_CATEGORY_FILTER";
