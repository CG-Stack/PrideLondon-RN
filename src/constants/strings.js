export default {
  venuedetailsGenderNeutralToilets: "Gender neutral toilets"
};
