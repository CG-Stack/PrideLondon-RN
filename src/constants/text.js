export default {
  tabEvents: "Events",
  filterByInterest: "I'm interested in...",
  eventDetailsPrice: "From: Â£",
  eventDetailsAccessibility: "Accessibility",
  eventDetailsAccessibilityDetails: "Accessibility Details",
  eventDetailsContact: "Contact",
  eventDetailsBuyButton: "Buy Tickets",
  eventDetailsGenderNeutralToilets: "Gender neutral toilets"
};
