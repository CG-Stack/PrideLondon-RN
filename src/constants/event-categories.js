// @flow
// Ideally the keys here should be consts, but currently these are setup in
// contentful only as the label, hence we need to use this for the key too.
// If we move to realise categories as its own content type then this will
// go away and we will fetch this data.
import * as colors from "./colors";
import type { EventCategoryMap } from "../data/event-deprecated";

const categories: EventCategoryMap = {
  "en-GB": {
    "Cabaret and Variety": {
      label: "Cabaret and Variety",
      color: colors.coralColor,
      contrast: false
    },
    Community: {
      label: "Community",
      color: colors.brightLightBlueColor,
      contrast: true
    },
    "Talks and Debates": {
      label: "Talks and Debates",
      color: colors.darkSkyBlueColor,
      contrast: false
    },
    "Film and Screenings": {
      label: "Film and Screenings",
      color: colors.lightTealColor,
      contrast: true
    },
    "Plays and Theatre": {
      label: "Plays and Theatre",
      color: colors.warmPinkColor,
      contrast: false
    },
    "Social and Networking": {
      label: "Social and Networking",
      color: colors.turquoiseBlueColor,
      contrast: true
    },
    Nightlife: {
      label: "Nightlife",
      color: colors.yellowColor,
      contrast: true
    },
    "Exhibition and Tours": {
      label: "Exhibition and Tours",
      color: colors.brightPurpleColor,
      contrast: false
    },
    "Sports and Activities": {
      label: "Sports and Activities",
      color: colors.vomitYellowColor,
      contrast: true
    },
    Health: {
      label: "Health",
      color: colors.bubblegumPinkColor,
      contrast: true
    },
    Music: {
      label: "Music",
      color: colors.cornflowerBlueColor,
      contrast: false
    }
  }
};

export default categories;
