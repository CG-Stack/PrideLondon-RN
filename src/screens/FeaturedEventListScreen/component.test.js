// @flow
import React from "react";
import type { NavigationScreenProp } from "react-navigation";
import { shallow } from "enzyme";
import Component from "./component";
import EventList from "../../components/EventList";
import { EVENT_DETAILS } from "../../constants/routes";

const navigation: NavigationScreenProp<{
  params: { title: string }
}> = ({
  navigate: jest.fn(),
  state: {
    params: {
      title: "Events"
    }
  }
}: any);

describe("FeaturedEventListScreen Component", () => {
  it("renders correctly", () => {
    const output = shallow(
      <Component
        navigation={navigation}
        events={[]}
        getAssetById={() => ({}: any)}
      />
    );
    expect(output).toMatchSnapshot();
  });

  it("renders navigation header correctly", () => {
    const params = Component.navigationOptions({ navigation });
    expect(params.title).toEqual("Events");
  });

  it("navigates to event details on event item press", () => {
    const output = shallow(
      <Component
        navigation={navigation}
        events={[]}
        getAssetById={() => ({}: any)}
      />
    );
    const onPress = output.find(EventList).prop("onPress");
    onPress("my-event");
    expect(navigation.navigate).toHaveBeenCalledWith(EVENT_DETAILS, {
      eventId: "my-event"
    });
  });
});

afterEach(() => {
  navigation.navigate.mockClear();
});
