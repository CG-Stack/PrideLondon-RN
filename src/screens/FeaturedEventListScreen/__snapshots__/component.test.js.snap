// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedEventListScreen Component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Header>
    <ContentPadding
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <IconButton
        accessibilityLabel="Back"
        onPress={[Function]}
        source={1}
      />
      <Text
        allowFontScaling={true}
        markdown={false}
        onLayout={[Function]}
        style={
          Object {
            "color": "#ffffff",
          }
        }
        type="h2"
      >
        Featured events
      </Text>
      <View
        style={
          Object {
            "height": 48,
            "width": 48,
          }
        }
      />
    </ContentPadding>
  </Header>
  <EventList
    events={Array []}
    getAssetUrl={[Function]}
    locale="en-GB"
    onPress={[Function]}
    refreshing={false}
  />
</View>
`;
